### PR TITLE
MTC Service Alerts: Change Character limit validation

### PR DESCRIPTION
**Pull request by robertgregg3**
### Character limit/validation on alert title and description
- Changed Character limit to count up rather than down
- Text to go red and show error when over the character limit
- Removed the popup validation so that the alert saves even if over the character limit

Referencing issue: https://github.com/ibi-group/datatools-ui/issues/647

### CODE REVIEW

**Review by evansiroky — CHANGES_REQUESTED** [review]
See review comments.

Also, please reference the relevant issue in the PR description.

File: lib/alerts/components/AlertEditor.js
Comment (landonreed) [reply lib/alerts/components/AlertEditor.js:296]: Also the `1200` number in the warning text should use the `ALERT_DESCRIPTION_CHAR_LIMIT` variable, rather than be hardcoded.

File: lib/alerts/components/AlertEditor.js
Comment (landonreed) [reply lib/alerts/components/AlertEditor.js:13]: Also, just a reminder to probably turn off your IDE's auto-reformatting of files. Generally we like to limit changes in PRs to the affected lines of code (rather than modifying a bunch of stuff wholesale).

File: lib/alerts/components/AlertEditor.js
Comment (landonreed) [reply lib/alerts/components/AlertEditor.js:208]: Yea, I think this is a mistake. We should just keep the danger class as it was (showing only if chars remaining is greater than `CHAR_WARNING_LIMIT`.

File: lib/alerts/components/AlertEditor.js
Comment (landonreed) [reply lib/alerts/components/AlertEditor.js:217]: As @evansiroky mentions in https://github.com/ibi-group/datatools-ui/pull/648/files/6a7d0de8c1f0b0f1a40428a4a93acbc8cf16bf2f#r599313022, we should move the entire `span` tag within the ternary statement:
```suggestion
                {titleCharacterCount > ALERT_TITLE_CHAR_LIMIT
                  ? <span className="text-danger">
                    WARNING: Alert title longer than{' '}
                    {ALERT_TITLE_CHAR_LIMIT} characters may get truncated in{' '}
                    some dissemination channels.
                  </span>
                  : null
                }
```

**Review by landonreed — CHANGES_REQUESTED** [review]
I left a few additional suggestions beyond what @evansiroky mentioned. In general, try to avoid modifying lines that are out of scope of the changes (I think this is just the IDE getting in the way!).

**Review by evansiroky — CHANGES_REQUESTED** [review]
In addition to the other changes requested, could you please also merge the `dev` branch into this branch? The `dev` branch now contains some code that enables GitHub Actions to run.

**Comment by robertgregg3** [comment]
Thanks for the feedback and comments. Yes a lot of the random movements quirkiness was the IDE.  Will take a close look at the comments and get on it right away.  I can see I created my Pull request way too early. So apologies for having to trawl through extra code that was unnecessary.

File: lib/alerts/components/AlertEditor.js
Comment (robertgregg3) [reply lib/alerts/components/AlertEditor.js:13]: I had a "sort imports" plugin from ITMS running amok.... it has been disabled.  Just so you know I wasn't randomly moving lines. I should have double checked.

Comment (landonreed) [reply lib/alerts/components/AlertEditor.js:13]: No worries at all

File: lib/alerts/components/AlertEditor.js
Comment (robertgregg3) [reply lib/alerts/components/AlertEditor.js:208]: Sure I will make that change.  

@landonreed just to confirm that we want the only form of warning that they are over the limit is the alert text and not a change of colour to the text?

**Review by landonreed — CHANGES_REQUESTED** [review]
@robertgregg3, if you check out [the changes on GitHub](https://github.com/ibi-group/datatools-ui/pull/648/files) you'll see a number of linting errors. A good way to check that the code looks good before committing is to run `yarn test` or `yarn lint` (if you want to skip the unit tests). Also, you might need to tell your IDE where to find the project's eslint config file to make sure it's configured properly (I'll see if Binh can help with that).

File: lib/alerts/components/AlertEditor.js
Comment (binh-dam-ibigroup) [inline lib/alerts/components/AlertEditor.js:233]: I will let @landonreed weigh on the final appearance, but could you see whether you could use some of the built-in functionality to display validation errors? See https://react-bootstrap-v3.netlify.app/components/forms/#forms-validation for the version of Bootstrap used for this repo.

Comment (binh-dam-ibigroup) [reply lib/alerts/components/AlertEditor.js:233]: If you do decide to use the layout as written, I would at least place the error message on a separate line. It is easy to miss otherwise given the low color contrast between dark red and gray.

File: lib/alerts/components/AlertEditor.js
Comment (binh-dam-ibigroup) [inline lib/alerts/components/AlertEditor.js:320]: I personally would place the error message below the input box.

Same remark regarding https://react-bootstrap-v3.netlify.app/components/forms/#forms-validation, and it can probably accomplished without using a `<h5>` heading.

**Review by binh-dam-ibigroup — CHANGES_REQUESTED** [review]
The main thing is to reformat the code so that only the actual changes appear, it makes reviews much easier. As noted, I would try to give a shot at using the react-bootstrap validation features (but I defer to @landonreed for final decisions).

File: lib/alerts/components/AlertEditor.js
Comment (binh-dam-ibigroup) [inline lib/alerts/components/AlertEditor.js:214]: No need for this extra indent.

File: lib/alerts/components/AlertEditor.js
Comment (binh-dam-ibigroup) [inline lib/alerts/components/AlertEditor.js:227]: Suggestion:
```js
{titleCharacterCount > ALERT_TITLE_CHAR_LIMIT && (<span>...)}
```

File: lib/alerts/components/AlertEditor.js
Comment (binh-dam-ibigroup) [inline lib/alerts/components/AlertEditor.js:312]: Same suggestion to use `&&` as above.

**Review by binh-dam-ibigroup — APPROVED** [review]
See my formatting suggestions, but looks good otherwise!

File: lib/alerts/components/AlertEditor.js
Comment (landonreed) [reply lib/alerts/components/AlertEditor.js:227]: I think the ternary is OK (we use both ternary statements and `&&` throughout our code), but the part after the `:` should be `null` not `''` (empty string literal).

File: lib/alerts/components/AlertEditor.js
Comment (landonreed) [inline lib/alerts/components/AlertEditor.js:27]: Generally we keep a new line between imports and stuff defined in the file.

File: lib/alerts/components/AlertEditor.js
Comment (landonreed) [inline lib/alerts/components/AlertEditor.js:230]: There's no need to encapsulate this in a string/curly braces. That's for evaluating javascript, but if you have a bunch of string literals you want to render, you can just place them directly in the jsx.
```suggestion
                                WARNING:  Alert title longer than ${ALERT_TITLE_CHAR_LIMIT} characters may get truncated in some dissemination channels.<br />
```

File: lib/alerts/components/AlertEditor.js
Comment (evansiroky) [reply lib/alerts/components/AlertEditor.js:227]: I don't really see a point in using a ternary when nothing is shown in the false condition. I recommend using @binh-dam-ibigroup's suggestion.

**Review by evansiroky — APPROVED** [review]
Approved pending resolution of remaining comments that Landon and Binh have made.

**Comment by evansiroky** [comment]
FYI, the failing GitHub Actions runs should be fixed with the changes proposed in #653.

File: lib/alerts/components/AlertEditor.js
Comment (robertgregg3) [reply lib/alerts/components/AlertEditor.js:227]: Good points. I think moving with more modern approaches is better and is less code too. Have changed it to &&

**Comment by landonreed** [comment]
@robertgregg3, we got some feedback from the client (see below). Feel free to go ahead with increasing the height of that input to match their request.

image

**Review by evansiroky — CHANGES_REQUESTED** [review]
Please merge `dev` into this branch as it has a fix that should make the CI checks work properly again.

**Comment by robertgregg3** [comment]
Have merged with dev and added the textarea styles

**Review by landonreed — APPROVED** [review]
Looks great! Thanks @robertgregg3!

**Pull request merged**
